core/sql/exp/ExpLOBaccess.cpp

```diff
@@ -1693,7 +1693,9 @@ Ex_Lob_Error ExLob::readDataToLocalFile(char *fileName,  Int64 offset, Int64 siz
     // open the targte file for writing 
     int filePerms = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
     int openFlags = O_RDWR ; // O_DIRECT needs mem alignment
-    if ((LobTgtFileFlags)fileflags == Lob_Append_Or_Error )
+    if (((LobTgtFileFlags)fileflags == Lob_Append_Or_Error ) ||
```

Comment:
Declaring fileflags to be of type LobTgtFileFlags would eliminate the need for all this casting. And would be safer. (Not a show-stopper though.)